cloudpickle/cloudpickle.py

```diff
@@ -268,9 +275,11 @@ def func():
                 # sys.modules.
                 if name is not None and name.startswith(prefix):
                     # check whether the function can address the sub-module
-                    tokens = set(name[len(prefix):].split('.'))
+                    submodule_relative_path = name[len(prefix):]
+                    tokens = set(submodule_relative_path.split('.'))
                     if not tokens - set(code.co_names):
-                        subimports.append(sys.modules[name])
+                        subimports.append(
+                            _getattribute(x, submodule_relative_path)[0])
```

Comment:
It could be the case that in `a.b.c`, module `c` would have been imported and therefore `a.b.b` would be found in `sys.modules` but somewhere in the code, someone would have done something like:

```python
import a.b
del a.b.c
```
and therefore the `_getattribute` would not find it.

Maybe we should fallback to the `sys.modules` lookup in this case.